atc/api/accessor/accessor_test.go

```diff
@@ -1,900 +1,649 @@
 		})
 
-		Context("when request has token", func() {
+		Context("when the user has no token", func() {
 			BeforeEach(func() {
-				token := jwt.NewWithClaims(jwt.SigningMethodRS256, claims)
-				tokenString, err := token.SignedString(key)
-				Expect(err).NotTo(HaveOccurred())
-				req.Header.Add("Authorization", fmt.Sprintf("BEARER %s", tokenString))
+				verification.HasToken = false
 			})
 
-			It("returns true", func() {
-				Expect(access.HasToken()).To(BeTrue())
+			It("returns false", func() {
+				Expect(result).To(BeFalse())
 			})
 		})
 
-		Context("when request has garbage token", func() {
+		Context("when the user has invalid token", func() {
 			BeforeEach(func() {
-				req.Header.Add("Authorization", fmt.Sprintf("BEARER %s", "29384q29jdhkwjdhs"))
+				verification.HasToken = true
+				verification.IsTokenValid = false
 			})
 
-			It("returns true", func() {
-				Expect(access.HasToken()).To(BeTrue())
+			It("returns false", func() {
+				Expect(result).To(BeFalse())
 			})
 		})
 
-		Context("when request has empty token", func() {
+		Context("when the user has valid token", func() {
 			BeforeEach(func() {
-				req.Header.Add("Authorization", fmt.Sprintf("BEARER %s", ""))
+				verification.HasToken = true
+				verification.IsTokenValid = true
+				verification.Claims = map[string]interface{}{
+					"federated_claims": map[string]interface{}{
+						"connector_id": "some-connector",
+						"user_id":      "some-user-id",
+					},
+				}
 			})
 
-			It("returns true", func() {
-				Expect(access.HasToken()).To(BeTrue())
-			})
-		})
+			Context("when the user is part of any admin team", func() {
+				BeforeEach(func() {
+					fakeTeam1.NameReturns("not-some-team")
+					fakeTeam1.AdminReturns(true)
+					fakeTeam1.AuthReturns(atc.TeamAuth{
+						"owner": map[string][]string{
+							"users": []string{"some-connector:some-user-id"},
+						},
+					})
+				})
 
-		Context("when request has no token", func() {
-			BeforeEach(func() {
-				req.Header.Del("Authorization")
+				It("returns true", func() {
+					Expect(result).To(BeTrue())
+				})
 			})
 
-			It("returns false", func() {
-				Expect(access.HasToken()).To(BeFalse())
-			})
-		})
-	})
+			Context("the team has the user configured as a 'member'", func() {
 
-	Describe("Is authenticated", func() {
-		JustBeforeEach(func() {
-			token := jwt.NewWithClaims(jwt.SigningMethodRS256, claims)
-			tokenString, err := token.SignedString(key)
-			Expect(err).NotTo(HaveOccurred())
-			req.Header.Add("Authorization", fmt.Sprintf("BEARER %s", tokenString))
-			access = accessorFactory.Create(req, "some-action")
-		})
-		Context("when valid token is set", func() {
-			It("returns true", func() {
-				Expect(access.IsAuthenticated()).To(BeTrue())
+				BeforeEach(func() {
+					fakeTeam1.NameReturns("some-team")
+					fakeTeam1.AuthReturns(atc.TeamAuth{
+						"member": map[string][]string{
+							"users": []string{"some-connector:some-user-id"},
```

Comment:
Should we also have tests for actions that require a specific role with TeamAuth using groups rather than users?

Reply:
done